Feature/url slugs

santiagogustavo opened the pull request:
Implementa a funcionalidade de slugs para as URLs de posts e categorias.
Utiliza o gem FriendlyId para tratar rotas com ids, títulos (para posts) e nomes (para categorias).
Atualiza todos os pontos onde esses modelos são referenciados e tratados pelos controllers :)

Commit:
(+) Adds FriendlyId to Gemfile and uses it for elegant slugs in Post model

- Adds FriendlyId to Gemfile, generates its model and includes it in the Post
model, using :title as the slug

- Updates Post model with :slug column, using null: false

- Updates documentation for Models (avoids Rubocop warnings)

Commit:
(+) Adds FriendlyId to Category model, updates model references and controller

- Updates Category model with FriendlyId and uses name as its slug

- Updates references to Category in Post controller and partials, so it treats correctly
the .find method calls and attributes usage

- Removes redundant variables in partials and controllers

Commit:
(~) Bugfixes in post creation and editing slug update

- Overrides the slug update function, so it can properly regenerate
Post slugs on edit

- Fixes a form error when creating and editing Posts (:category_id was
not present, took it out by mistake in previous commits)

guilpejon commented on blogster/app/controllers/comments_controller.rb:
```diff
@@ -38,7 +38,7 @@ def update
   # Obtains the correct Comment with before_action and destroys it
   def destroy
-    @fallback = post_path(@comment.post_id)
+    @fallback = post_path(@comment.post)
```
Se você não tiver usando `@fallback` na view você não precisa declarar como variável global aqui. Se não tiver usando essa variável também você pode passar direto o valor dela pro `redirect_back`, aí fica mais limpo o código.

santiagogustavo replied:
Coloquei essa variável por conta do @comment.destroy. A ideia seria voltar para o post, mesmo se tiver algum problema e cair no fallback

guilpejon commented on blogster/app/controllers/posts_controller.rb:
```diff
@@ -40,6 +40,7 @@ def create
   def edit; end
 
   def update
+    @post.slug = nil
```
Por que você tá setando o slug como nil aqui?

santiagogustavo replied:
O slug não atualizava quando um post era editado, problema relacionado com o regenerate_slug padrão do FriendlyId. Fiz um override no modelo, vou verificar se continua funcionando sem essa atribuição para nil.

Commit:
(~) Removes redundant code and unnecessary method atributions

- In Post controller, there was a line used to set slug to nil
in order to fix the problem of slugs not updating. Since a more
elegant solution was already defined in Post model, there was no
need to keep this attribution

- In Comment controller, there was a variable @fallback used with
post path storing purpose, later used in redirect_back. This one
was removed and the fallback path is now the root path

santiagogustavo commented:
@guilpejon fiz os ajustes que foram comentados, não fazia sentido manter aquelas linhas mesmo
